### DISCUSSION TITLE
Scenario "Jakarta": it's always DNS.

### DISCUSSION THREAD

**Discussion by fduran** [SadServers/sadservers#38] [Scenarios]
Discussions about scenario [Jakarta](https://github.com/fduran/sadservers/tree/main/scenarios/jakarta)

**Comment by norcalipa** [comment]
I got the /home/ubuntu/agent/check.sh script returning OK, but I can't get credit because the "Check My Solution" button doesn't run it and check the output.  Perhaps it needs to be moved to /home/admin/agent like other scenarios use?  Not sure as I don't know how these scenarios are set up, but this appears to be different compared to scenarios that work properly.

**Comment by fduran** [comment]
Hello @norcalipa , I can add a proper check.sh to this scenario (or take out the current one since there's no "Check My Solution" button) but credit is given (see your scenarios page for ex) to the two scenarios that don't have a checker if they are tried.

**Comment by sabir69261** [comment]
In this scenario, we are looking for a DNS solution because, as soon as google.com resolves, the problem is considered solved. However, in check.sh, we are currently checking whether ping is successful. Unfortunately, ping will not succeed because there is no connectivity to the outside internet. We need to change the if condition in check.sh as shown below.

if [[ "$res" = "0 received, 100% packet loss" ]]

**Comment by Sharpie** [comment]
The shell command `getent hosts google.com` might also work as a check. I think that forces a lookup through NSSwitch.
